TVC_MVP/CountyPresenter.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  CountyPresenter.swift
+//  TVC_MVP
+//
+//  Created by Shane McCully on 28/02/2019.
+//  Copyright © 2019 Shane McCully. All rights reserved.
+//
+
+import Foundation
+
+private struct Constants {
+
+    static let alertTitleBlock = "County ID for "
+    static let alertTitleBlockEnd = " is"
+    static let counties = "counties"
+    static let json = "json"
+    static let name = "name"
+    static let id = "id"
+    static let jsonError = "Something went wrong"
+    static let tryAgain = "Please try again"
+
+}
+
+class CountyPresenter: CountyPresnterProtocol {
+
+    unowned let view: CountyViewProtocol
+    var countyArray: [County] = [County]()
+
+    // MARK: - Initialization
+
+    required init(view: CountyViewProtocol) {
+        self.view = view
+    }
+
+    // MARK: - CountyPresnterProtocol methods
+
+    func viewDidLoad() {
+        parseJSON()
+    }
+
+    func reloadData() {
+        view.reloadData()
+    }
+
+    func numberOfSections() -> Int {
+        return 1
+    }
+
+    func numberOfRowsInSection() -> Int {
+        return countyArray.count
+    }
+
+    func removeCounty(at indexPath: IndexPath) {
+        countyArray.remove(at: indexPath.row)
+        reloadData()
+    }
+
+    func fetchCounty(for indexPath: IndexPath) -> County? {
+        return countyArray[indexPath.row]
+    }
+
+    func presentAlert(at indexPath: IndexPath) {
+        let dismiss = AlertAction.DefaultActions.dismissAction()
+        let delete = AlertAction.DefaultActions.deleteAction {
+            self.removeCounty(at: indexPath)
```

Comment:
You did it properly on the other block but not on this one, there is danger of a retain cycle happening.